### PR TITLE
Add serve command 

### PR DESCRIPTION
**Pull request by arslee07**

### CODE REVIEW

File: cli/pubspec.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ dependencies:
   markdown: ^4.0.0
   mustache_template: ^2.0.0
   path: ^1.8.0
+  shelf_static: ^1.1.0
```
Comment (HarryET) [inline cli/pubspec.yaml:16]: Sort formatting

**Review by HarryET — CHANGES_REQUESTED** [review]
Looks at comments please

File: cli/pubspec.yaml (hunk 1 above)
Comment (arslee07) [reply cli/pubspec.yaml:16]: I'm not quite sure what you mean?

Comment (l7ssha) [reply cli/pubspec.yaml:16]: deps should be sorted alphabethically

File: cli/lib/commands/ServeCommand.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class ServeCommand extends Command {
+  @override
+  String get description => "Run the builtin development server";
+
+  @override
+  String get name => "serve";
+
+  ServeCommand() {
+    argParser.addOption("port", defaultsTo: "3000", valueHelp: "3001");
```
Comment (l7ssha) [inline cli/lib/commands/ServeCommand.dart:11]: any reason behind choosing 3000 port?

File: cli/lib/commands/ServeCommand.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    );
+    final port = int.tryParse(argResults!["port"] as String)!;
+
+    final handler = shelf_static.createStaticHandler(
+      destinationDir,
+      defaultDocument: "index.html",
+    );
+
+    await shelf_io
+        .serve(handler, "localhost", port)
```
Comment (l7ssha) [inline cli/lib/commands/ServeCommand.dart:29]: if we are going to allow spcifying port maybe good idea would be host too?

**Review by l7ssha — CHANGES_REQUESTED** [review]

File: cli/lib/commands/ServeCommand.dart (hunk 2 above)
Comment (arslee07) [reply cli/lib/commands/ServeCommand.dart:11]: > any reason behind choosing 3000 port?

nope

File: cli/lib/commands/ServeCommand.dart (hunk 3 above)
Comment (arslee07) [reply cli/lib/commands/ServeCommand.dart]: done

File: cli/pubspec.yaml (hunk 1 above)
Comment (arslee07) [reply cli/pubspec.yaml:16]: is it already sorted, or am I missing something?

Comment (HarryET) [reply cli/pubspec.yaml:16]: Don't worry. Its resolved.

File: cli/lib/commands/ServeCommand.dart (hunk 2 above)
Comment (l7ssha) [reply cli/lib/commands/ServeCommand.dart:12]: Good enough for me

**Review by l7ssha — APPROVED** [review]

**Review by HarryET — APPROVED** [review]

**Pull request merged**